### PR TITLE
fix bug on colocate agg (name error)

### LINKED ISSUE #6113: [Bug] 表名影响了colocate agg
**Describe the bug**
colocate agg判断的时候，如果Group by字段带上表名, 则不会实现colocate agg。

**To Reproduce**
* explain select count(1) from a join b on a.k1 = b.k1 group by a.k1;
* explain select count(1) from a group by a.k1;
* 
**Expected behavior**
计划中Group by不需要做merge finalize

**Screenshots**
表a、b的结构如下：
image
image

不加表名的Group by可以正常work：
image

加了表名之后不能优化成colocate agg:
image

colocate join加Group by中包含表名也是如此：
image

### PR DESCRIPTION
**Pull request by Userwhite**
## Proposed changes

修改了https://github.com/apache/incubator-doris/issues/6113 https://github.com/apache/incubator-doris/issues/6066 提出的bug 
由于表名有的获取了db，有的没有，所以实际一致的表名会被当作不等。
判断表名的时候，都获取完整的表名即可(Group by字段没有表名的时候会在columnEqual的时候去获取表名，但有表名的时候不会，而这个表名不带集群，显然和dataPartition的表名字不一致)

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #6113 #6066) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/test/java/org/apache/doris/planner/QueryPlanTest.java [hunk 1]
```diff
@@ -1360,6 +1376,32 @@ public void testAggregateSatisfyOlapTableDistribution() throws Exception {
         explainString = UtFrameUtils.getSQLPlanOrErrorMsg(connectContext, "EXPLAIN " + sql);
         System.out.println(explainString);
         Assert.assertTrue(explainString.contains("AGGREGATE (update serialize)"));
+    
+        // 测试colocate join是否能够colocate agg
```
Comment (morningman) [inline fe/fe-core/src/test/java/org/apache/doris/planner/QueryPlanTest.java:1380]: Thanks for your PR, but please using english to comment

Comment (Userwhite) [reply fe/fe-core/src/test/java/org/apache/doris/planner/QueryPlanTest.java]: fixed

Comment (Userwhite) [reply fe/fe-core/src/test/java/org/apache/doris/planner/QueryPlanTest.java]: @morningman  Hi~ see this pr if have free time～

File: fe/fe-core/src/main/java/org/apache/doris/analysis/SlotRef.java [hunk 2]
```diff
@@ -134,8 +134,16 @@ public boolean columnEqual(Expr srcExpr) {
             srcTableName = srcSlotRef.getTableName();
         }
         TableName thisTableName = tblName;
-        if (thisTableName == null && desc != null) {
-            thisTableName = getTableName();
+        if (desc != null && desc.getParent() != null && desc.getParent().getRef() != null) {
+            thisTableName = desc.getParent().getRef().getName();
+        } else if (!desc.getSourceExprs().isEmpty()) {
+            List<Expr> sourceExprs = desc.getSourceExprs();
+            Expr sourceExpr = sourceExprs.get(0);
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/SlotRef.java:141]: source exprs get (0) ? how about others?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/SlotRef.java [hunk 3]
```diff
@@ -134,8 +134,16 @@ public boolean columnEqual(Expr srcExpr) {
...
+            if (((SlotRef) sourceExpr).desc != null && ((SlotRef) sourceExpr).desc.getParent() != null) {
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/SlotRef.java:142]: This value does not have to be slotREF

File: fe/fe-core/src/main/java/org/apache/doris/analysis/SlotRef.java (hunk 2 above)
Comment (Userwhite) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/SlotRef.java:141]: When slotref can't get db.table from "desc.getParent().getRef().getName", I think the sourceExpr of partitionExprs is only one slotRef. The union situation maybe happen? 
Or I check all sourceExpr's tablename, if them is the same, then it's the actual table name.

File: fe/fe-core/src/main/java/org/apache/doris/analysis/SlotRef.java (hunk 3 above)
Comment (Userwhite) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/SlotRef.java:142]: If sourceExpr didn't instanceof SlotRef, I can't get db.name(default_cluster:db_name.table_name), is it right?

Comment (EmmyMiao87) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/SlotRef.java:142]: But source expr may not be slotref, it may be an expression of a+b

**Pull request closed without merging**